packages/go/index.ts

```diff
@@ -289,18 +289,20 @@ export async function build({
     // we need our `main.go` to be called something else
     const mainGoFileName = 'main__vc__go__.go';
 
-    if (packageName !== 'main') {
-      const go = await createGo(
-        workPath,
-        goPath,
-        process.platform,
-        process.arch,
-        {
-          cwd: entrypointDirname,
-          stdio: 'inherit',
+    const go = await createGo({
+      goPath,
+      modulePath: goModPath,
+      opts: {
+        cwd: entrypointDirname,
+        env: {
+          GOARCH: 'amd64',
+          GOOS: 'linux',
```

Comment:
Can we use `process.platform` and `process.arch` instead so this will always download the correct version?

Reply:
We don't want to use the local machine's platform/arch. This is in the `build()` function where we want to force the `GOARCH` and `GOOS` to the platform and architecture that AWS Lambda supports. This allows us to `vc build && vc deploy` from any device (macOS/Linux/Windows) regardless of architecture (amd64/arm64).

Reply:
@cb1kenobi Did you actually test that hypothesis? I wouldn't expect a linux build to work on mac or windows when running `vc build`.

Reply:
It's not a hypothesis. Go can build a native binary for any platform/arch. Yes I tested it on both Windows (amd64) and macOS (arm64). I also tested the amd64 version of Go on mac which Rosetta will virtualize and build just fine with only a little overhead.